lib/src/str_util.rs

```diff
@@ -50,6 +50,19 @@ impl StringPattern {
         StringPattern::Substring(String::new())
     }
 
+    /// Parses the given string as a `StringPattern`. Everything before the
+    /// first ":" is considered the string's prefix. If the prefix is "exact:",
+    /// "glob:", or "substring:", a pattern of the specified kind is returned.
+    /// Returns an error if the string has an unrecognized prefix. Otherwise, a
+    /// `StringPattern::Exact` is returned.
+    pub fn parse(src: &str) -> Result<StringPattern, StringPatternParseError> {
```

Comment:
This could be `FromStr` (to let clap automatically choose the parsing function), but that can be decided later. I'm not sure if this format  should be the canonical string representation.

(no action needed)

Reply:
Sounds good, I wasn't sure if we wanted this to be the canonical string rep either. Also SGTM about not adding the advance-branches config to UserSettings. Maybe it would be better to parse them in the cli crate.